packages/react/src/primitives/shared/styleUtils.ts

```diff
@@ -167,31 +167,47 @@ export const useStyles = (
 
 interface SplitProps<PrimitiveProps> {
   styleProps: AllStyleProps;
-  rest: Omit<
-    PrimitiveProps,
-    keyof FlexContainerStyleProps | keyof BaseStyleProps
+  rest: Partial<
+    Omit<PrimitiveProps, keyof FlexContainerStyleProps | keyof BaseStyleProps>
   >;
 }
 
+function isStyleKey(prop: any): prop is keyof AllStyleProps {
+  if (prop in ComponentPropsToStylePropsMap) {
```

Comment:
Would just returning `prop in ComponentPropsToStylePropsMap` also work? Seems like that should already return a boolean.